### ISSUE TITLE
Is there anyway to do self relationships?

### ISSUE THREAD

**Issue by cdrandin** [vuex-orm/vuex-orm#430]
I have a comments model like this

```javascript

export default class Comment extends Model {
    static entity = "comments";

    static fields() {
        return {
            id: this.attr(null),
            user_id: this.attr(null),
            level: this.attr(0),
            parent: this.attr(null),
            comment: this.attr(""),
            // children: this.hasMany(Comment, "id"),
            user: this.belongsTo(User, "user_id"),
            created: this.attr(""),
            modified: this.attr(""),
        };
    }
}
```

I am trying to get a nested relationship to work with the same model. Currently, this way doesn't seem to do it or possibly I am doing it incorrectly.

**Comment by tvillaren** [comment]
Hello @cdrandin,

The same way as you need a foreign key (FK) to reference the User (through the `user_id` relationship), you also need a FK on Comment to reference its "parent" Comment. 
This is actually the key you need to pass in the field definition:

```js
// ...
return {
  id: this.attr(null),
  parent_id: this.attr(null),
  children: this.hasMany(Comment, 'parent_id')
  // ...
};
```

**Comment by kiaking** [comment]
Hmm... strange. We do have a test case for these kind of same entity nested relationship. Is it possible for you to create reproducible code on this so we can debug farther?

**Comment by kiaking** [comment]
@cdrandin Oh OK, so you're trying to insert all relationship children, right? Maybe it is related to #419. Which is that when you use `where` then the relationship is not inserted. Could you try including id (where value) to the `data` property and see what happens?

**Comment by cdrandin** [comment]
@kiaking I believe I did what you requested if not let me know.

I went from 

```javascript
Comment.update({
        where: comment_id,
        data: {
          comment: "override me",
          children
        }
      });
```

to 

```javascript
Comment.update({
        where: comment_id,
        data: {
          id: comment_id,
          comment: "override me",
          children
        }
      });
```

The result is still the same.

**Comment by kiaking** [comment]
@cdrandin Oh sorry, you need to remove `where` completely. Like this.

```js
Comment.update({
        data: {
          id: comment_id,
          comment: "override me",
          children
        }
      });
```

**Comment by cdrandin** [comment]
@kiaking Still the same.

**Comment by cdrandin** [comment]
Something I also am noticing.

In the codesandbox above in mounted. When I use the the model object to find my comment I see `user` which is supposed to be an object and it is there, however, when I check the state directly it is always null.

Is this intentional?

**Comment by kiaking** [comment]
Hmm.. I guess I need to look into this a bit more.

> In the codesandbox above in mounted. When I use the model object to find my comment I see user which is supposed to be an object and it is there, however, when I check the state directly(this.$store.state.entities.comments.data) it is always null.

Yes, this is intended. Relationships are loaded when you fetch records.

**Comment by cdrandin** [comment]
I am not entirely sure what is going on here.

```javascript
CommentModel.create({
        data: {
          comment: "inserted comment",
          children: [
            {
              id: "12312312312321",
              comment: "hello",
              user: {
                id: "1b81028d-6c29-4d64-94a3-d8f4279aca74",
                username: "king",
                name: "king",
                profile: {
                  cover: null,
                  thumbnail: "http://localhost:8000/media/default-avatar.jpg",
                  is_private: true,
                  created: "2019-05-09T15:36:34-0700"
                }
              }
            }
          ],
          user: {
            id: "1b81028d-6c29-4d64-94a3-d8f4279aca74",
            username: "king",
            name: "king",
            profile: {
              cover: null,
              thumbnail: "http://localhost:8000/media/default-avatar.jpg",
              is_private: true,
              created: "2019-05-09T15:36:34-0700"
            }
          }
        }
      })
```

I check the state using the query builder and I get that both items were entered into vuex-orm, yet children is still empty. 

The funny, not so funny, the thing is when I initially load my data I get children populated correctly. Working just fine going down 4 levels too.


@kiaking I was trying to play with it more to see where this issue could be resolved. Do you have any places you'd recommend searching?

I found:

- `../src/attributes/relations/HasMany.ts`
- `../src/model/Model.ts`

P.S. 

I also think this issue is related to https://github.com/vuex-orm/vuex-orm/issues/123, which never got resolved.

**Comment by cdrandin** [comment]
@kiaking 	

Welp, I think I figured it out. It was my misunderstanding of [Insert Method for Relationships](https://vuex-orm.github.io/vuex-orm/guide/data/inserting-and-updating.html#insert-method-for-relationships). I had multiple models sharing a single model and when I created one the other groups of users would be erased.

I just didn't realize creating was clearing out my other models that happened to share a model.

Well, in any event, I was wondering if there could be maybe an improvement on usability for the insert methods for the relationship to include all models related.

Ex)

```javascript
User.create({
  data: [...],
  insert: ['posts', 'profile', 'friends', etc]
})
```

could be reduced to 

```javascript
User.create({
  data: [...],

  // which ever variation you prefer
  insert: ['all'],
  insert: ['$all'],
  insert: ['*'],
  insert: true,
  insertAll: true,
})
```
